### PR TITLE
feat(quantic): added localization for component labels

### PR DESCRIPTION
**Pull request by nathanlb**
This PR includes:
- Addition of Salesforce Custom Label definitions with corresponding french translations taken from Atomic locale string.
- Addition of a I18nService utility class for some label formatting.
- Replacement of strings for custom labels in Quantic components.

Exceptions:
- QuanticResult `Ignored` : Our current result template uses labels that won't be present in our template after the visual rework coming shortly.

French example-search
Screen Shot 2021-07-20 at 11 51 46 AM

### CODE REVIEW

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (jeremierobert-coveo) [inline packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:2]: After installing the unlocked package are users able to edit the labels ?

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.html
Comment (jeremierobert-coveo) [inline packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.html:28]: Those aria made me think. Do we have the capability to test automaticly for accessiblity ?
I know that is a requirement for atomic iirc

**Review by jeremierobert-coveo — COMMENTED** [review]
Some unit tests are missings but overral that looks great.

Have doc reviewed those strings ?

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.html
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.html:28]: We currently don't have it implemented but there's a [jira for that](https://coveord.atlassian.net/browse/SFINT-3969) and yes there are tests we can run to check for accessibility. Atleast to a certain degree.

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (nathanlb) [reply packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:2]: Yes they can edit the labels and translations

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js:72]: Yes you're right I will add a couple tests.

**Comment by nathanlb** [comment]
> Some unit tests are missings but overral that looks great.
> 
> Have doc reviewed those strings ?

I took the strings from Atomic so I assumed their doc approved in some way. They're probably taken from the JSUI tbh.

File: packages/quantic/force-app/main/default/lwc/quanticSummary/quanticSummary.js
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticSummary/quanticSummary.js:75]: Isn't the text cramped without the whitespaces?

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js]: Technically, zero is plural in english and singular in french. We could have a third variant (for zero/none) for these labels to handle all the pluralization in one place.

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js:70]: nice!

**Review by lbergeron — COMMENTED** [review]
Minor comments overall. Waiting to see mostly if pluralization changes or not before approving.

File: packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.html
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.html:6]: I wasn't exactly sure what to do here because this all label and button are not part of the redesigned category facets. But "All" didn't really describe what this was doing either so I switched it to clear. I'll follow your suggestion and change the icon for clear.

File: packages/quantic/force-app/main/default/lwc/quanticSummary/quanticSummary.js
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticSummary/quanticSummary.js:75]: This is according to the mock ups where I think it looks all right. You disagree? 
Screen Shot 2021-07-22 at 10 53 33 AM

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js]: You mean having another label with a `_zero` suffix? And apply that suffix here following a truthy comparison with 0?

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js:70]: Thanks!

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js]: That makes sense to me. But so far the zero isn't going to appear I think. I don't think it's possible for a query to return in zero seconds and if a query returns 0 results we'll show the no results component instead of the query summary.

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (erocheleau) [inline packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:60]: Why do we need a different one for 0 seconds? Also is 0 seconds even possible?

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js
Comment (lbergeron) [reply packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js]: If `getLabelNameWithCount` does not handle the zero case, then we'll have to handle it in every place where a pluralized string is used. We'll certainly forget to handle that in some places and it would lead to localization issues (granted those are minor issues). I'm just wondering if having `getLabelNameWithCount` handle this case would make it easier to maintain over time.

Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js]: I agree. I added the zero variants. now we just have to remember to create labels for all 3 scenarios when applicable 👍

File: packages/quantic/force-app/main/default/lwc/exampleSearch/exampleSearch.css
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/exampleSearch/exampleSearch.css:1]: @nathanlb so did you find a fix so the page isn't jumping in width whenever there's no result?

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (lbergeron) [inline packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:19]: Why not use `No results` instead?

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (lbergeron) [inline packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:40]: Why not use `No results for {{2}}` instead?

File: packages/quantic/force-app/main/default/lwc/quanticSummary/quanticSummary.js
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticSummary/quanticSummary.js:91]: Shouldn't you call `I18nService.getLabelNameWithCount('inSeconds', duration)` instead?

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js:64]: Well, I just found it is not perfect yet. When dealing with decimal numbers, pluralization differs between english and french, and most likely other languages as well.

In french, plural is applied to values greater or equal to 2. In english, plural is applied to any value different from 1.

https://www.antidote.info/fr/blogue/enquetes/15-gramme-ou-15-grammes#:~:text=En%20fran%C3%A7ais%2C%20seules%20les%20quantit%C3%A9s,prend%20la%20marque%20du%20pluriel.

Do you think the `@salesforce/i18n` module could help with that?

File: packages/quantic/force-app/main/translations/fr.translation-meta.xml
Comment (lbergeron) [inline packages/quantic/force-app/main/translations/fr.translation-meta.xml:12]: `Aucun résultat`

File: packages/quantic/force-app/main/translations/fr.translation-meta.xml
Comment (lbergeron) [inline packages/quantic/force-app/main/translations/fr.translation-meta.xml:24]: `Aucun résultat pour {{2}}`

File: packages/quantic/force-app/main/default/lwc/quanticBreadcrumbManager/quanticBreadcrumbManager.js
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticBreadcrumbManager/quanticBreadcrumbManager.js:13]: I like this it looks really clean!

File: packages/quantic/force-app/main/translations/fr.translation-meta.xml
Comment (nathanlb) [reply packages/quantic/force-app/main/translations/fr.translation-meta.xml:12]: lol yes. I didn't think about it much because it should never show up. you're right.

**Review by jeremierobert-coveo — APPROVED** [review]
Love the util

File: packages/quantic/force-app/main/default/lwc/quanticDateFacetValue/quanticDateFacetValue.js
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticDateFacetValue/quanticDateFacetValue.js:25]: Very very nice!

File: packages/quantic/force-app/main/default/lwc/quanticSearchBox/quanticSearchBox.js
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticSearchBox/quanticSearchBox.js:8]: I think I would create the `labels = { search }` even if there is just one, might be a bit confusing seeing only `${search}...` in the code.

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js:54]: I18nService? Or I18nUtils?

Just nitpicking on the name, you decide :d

**Review by erocheleau — APPROVED** [review]

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (nathanlb) [reply packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml]: yeah i'm dumb i just yeeted that one. The query summary shouldn't show up when there's no results but i changed it

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (nathanlb) [reply packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:32]: I copied what was already done in atomic. I don't know enough languages to be confident that the same sentence structure will apply all the time.

File: packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (nathanlb) [reply packages/quantic/force-app/main/default/labels/CustomLabels.labels-meta.xml:60]: No but @lbergeron pointed out that the rules for plurality with zero differ from one language to the next. so now it looks creates a `_zero` variant. We likely won't need this label ever but the util will return a label name that doesn't exist otherwise. It's not a perfect solution...

File: packages/quantic/force-app/main/default/lwc/exampleSearch/exampleSearch.css
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/exampleSearch/exampleSearch.css:1]: No but if it's ok with y'all I think it would be better to wait till we have all our search page components (including the NoResults and loading animation stencils, which should at least stop it from collapsing) before we attempt to fix it properly.

File: packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js:70]: The `@salesforce/i18n` module doesn't appear to be very powerful. Mostly date/currency type things.
I added a bit of logic to be able to define plurality. But I found that certain languages also have different types of plural. (ex: Arabic has 5)
So I'm wondering if this won't be able to scale properly. Maybe we should use an `i18next` module like atomic... But then I don't know if that'll work well with these salesforce custom labels.
What do you think?

Comment (lbergeron) [reply packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js:70]: Using `i18next` might help us. We'd have to try, but if we use `i18next` to get the suffix and Salesforce has a key with the right suffix, then it should just work, I guess.

Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js:70]: That would be ideal indeed. I'll look into it

**Review by lbergeron — APPROVED** [review]
I don't want to block the PR because of pluralization corner cases. We should have a look at `i18next` though as it could solve the issue.

File: packages/quantic/force-app/main/default/lwc/exampleSearch/exampleSearch.css
Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/exampleSearch/exampleSearch.css:1]: I propose we wait till we have all our basic components (including No Results and loading animation stencils) which should atleast prevent it from collapsing. Then I think we can best find the way to fix it.

**Pull request merged**
